### PR TITLE
move to GitHub Container Registry

### LINKED ISSUE #52: Docker is taking away free teams
Got this:

```
Docker is sunsetting Free Team organizations

Free Team organizations are a legacy subscription tier that no longer exists. This tier included many of the same features, rates, and functionality as a paid Docker Team subscription.

After reviewing the list of accounts that are members of legacy Free Team organizations, we’ve identified yours as potentially being one of them.

If you own a legacy Free Team organization, access to paid features — including private repositories — will be suspended on April 14, 2023 (11:59 pm UTC). Upgrade your subscription before April 14, 2023 to continue accessing your organization.
```

I think it might affect as @SaraSmiseth what do you think?

### PR DESCRIPTION
**Pull request by AndrewSav**
## Description

Since Docker discontinues free teams we need another place to store and manage our images. This PR moves it to GitHub container registry.

Fixes #52

## Type of change
- [X] Other... 

## Status
- [X] Ready

## Todo List
- [X] Implementation
- [X] Documentation

## How has this been tested ?

I pushed all 4 images to GitHub Container registry. I pulled them while being unauthenticated to make sure that public can access it

### CODE REVIEW

File: VERSIONING.md [hunk 1]
```diff
@@ -25,10 +25,25 @@ Follow these steps when an update is ready to be pushed to Docker Hub:
...
-docker push mailserver2/debian-mail-overlay:latest
+docker tag debian-mail-overlay ghcr.io/mailserver2/debian-mail-overlay:0.0.0
+docker push ghcr.io/mailserver2/debian-mail-overlay:0.0.0
+docker tag debian-mail-overlay ghcr.io/mailserver2/debian-mail-overlay:latest
+docker push ghcr.io/mailserver2/debian-mail-overlay:latest
+```
+
+## Authenticating with GitHub Container Registry
+
+Create a [personal access token](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-with-a-personal-access-token-classic) by going [here](https://github.com/settings/tokens/new?scopes=write:packages,delete:packages).
```
Comment (sknight80) [inline VERSIONING.md:40]: NIT: do we want to add:
`Make sure you set expiration date for your personal token`
?

**Comment by sknight80** [comment]
I also updated the following Wiki page: https://github.com/mailserver2/mailserver/wiki/Migration-from-hardware-mailserver

**Comment by AndrewSav** [comment]
> I also updated the following Wiki page: https://github.com/mailserver2/mailserver/wiki/Migration-from-hardware-mailserver

Thank you for that

File: VERSIONING.md (hunk 1 above)
Comment (AndrewSav) [reply VERSIONING.md:40]: I assumed this was mostly for the maintainers and the maintainers should ideally know good security practices. Right now it's just the three of us and I hope we are in agreement about that. If you think it's important enough I do not mind adding it, but I did consider it and decided that I should not be teaching you guys what your security practices should be ;)

**Comment by AndrewSav** [comment]
I put the changes into <https://github.com/mailserver2/mailserver/tree/ghcr> for safekeeping, I'm closing the PR now.

**Pull request closed without merging**